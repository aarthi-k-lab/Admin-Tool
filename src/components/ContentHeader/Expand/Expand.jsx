import React from 'react';
import PropTypes from 'prop-types';
import { IconButton } from '@material-ui/core';
import ZoomOut from '@material-ui/icons/ZoomOutMap';

const Expand = ({ onClick }) => (
  <IconButton onClick={onClick}>
    <ZoomOut />
  </IconButton>
);

Expand.defaultProps = {
  onClick: () => {},
};

Expand.propTypes = {
  onClick: PropTypes.func,
};

export default Expand;
