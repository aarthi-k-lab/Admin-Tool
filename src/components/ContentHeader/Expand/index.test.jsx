import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Expand from '.';

describe('<Expand />', () => {
  test('snapshot test', () => {
    const tree = renderer
      .create(<Expand />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('triggers onClick handler', () => {
    const handleClick = jest.fn();
    const wrapper = shallow(<Expand onClick={handleClick} />);
    wrapper.simulate('click');
    expect(handleClick.mock.calls.length).toBe(1);
  });
});
