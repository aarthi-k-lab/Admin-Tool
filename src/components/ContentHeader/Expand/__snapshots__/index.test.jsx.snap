// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Expand /> snapshot test 1`] = `
<button
  className="MuiButtonBase-root MuiIconButton-root"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiIconButton-label"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <defs>
        <path
          d="M0 0h24v24H0z"
          id="a"
        />
      </defs>
      <path
        d="M15 3l2.3 2.3-2.89 2.87 1.42 1.42L18.7 6.7 21 9V3zM3 9l2.3-2.3 2.87 2.89 1.42-1.42L6.7 5.3 9 3H3zm6 12l-2.3-2.3 2.89-2.87-1.42-1.42L5.3 17.3 3 15v6zm12-6l-2.3 2.3-2.87-2.89-1.42 1.42 2.89 2.87L15 21h6z"
      />
    </svg>
  </span>
</button>
`;
